tests/spec/puppeteer-spec.js

```diff
@@ -147,4 +150,22 @@ describe("Puppeteer basic tests for the Python Editor.", function() {
         expect(codeName).toEqual("main");
     });
 
+    it("Saves a python file with the correct filename", async function(){
+        const page = await global.browser.newPage();
+        await page._client.send('Page.setDownloadBehavior', {behavior: 'allow', downloadPath: './spec/test-files/temp-test-files'});
```

Comment:
It'd be good to ensure the `program_test.py` file is not present before running the test, in case it's a left over from previous test runs, in that case we would get a false pass.

Reply:
Added a check in case file already exists, and if does, it deletes it.